### DISCUSSION TITLE
How do I use the search method?

### DISCUSSION THREAD

**Discussion by sittelle** [lookfirst/sardine#365] [General]
Hi,

I would like to search an item by a specific QNAME value (e.g. by etag). I do not understand how to do this with the search method provided by sardine.

Thank you!

**Comment by lookfirst** [comment]
Sardine just passes through webdav commands. It doesn't actually do a search. That's up to the server.

I think your server would need to support this RFC: http://www.webdav.org/specs/rfc5323.html

I see in the sardine xsd that we have `searchrequest`, which is defined here: http://www.webdav.org/specs/rfc5323.html#rfc.section.2.3.3

I don't think we fully implemented that RFC though.

My suggestion for trying to find a resource via etag would just be to implement something RESTful outside of webdav and sardine actually. You're probably going to need additional metadata there too, so it is kind of outside the scope of this project.

We're focused mostly just on the file serving aspects of webdav.

**Comment by sittelle** [comment]
Yes, my question was aimed at Sardine.search method. Unfortunately, I do not understand what a valid query should like as it gets translated into xml in that method.
